clients/src/main/java/org/apache/kafka/common/security/plain/PlainSaslServer.java

```diff
@@ -155,7 +149,7 @@ public SaslServer createSaslServer(String mechanism, String protocol, String ser
             if (!PLAIN_MECHANISM.equals(mechanism)) {
                 throw new SaslException(String.format("Mechanism \'%s\' is not supported. Only PLAIN is supported.", mechanism));
             }
-            return new PlainSaslServer(cbh);
+            return new PlainSaslServer(cbh, new JaasPlainPasswordVerifier());
```

Comment:
This does not look pluggable to me : a concrete implementation of the password verifier is hardcoded rather than say instantiating the class from a config entry.